### PR TITLE
Fix: Add schedule button to dashboard page

### PR DESCRIPTION
**Pull request by Xuefeng-Zhu**

### CODE REVIEW

File: packages/dashboards/tests/integration/components/navi-dashboard-test.js [hunk 1]
```diff
@@ -115,4 +116,25 @@ test('dashboard export', function(assert) {
...
+    assert.ok($('.action.schedule').is(':visible'),
+      'Dashboard schedule button should be visible');
+
+    config.navi.FEATURES.enableScheduleDashboards = false;
+    this.render(hbs`{{navi-dashboard dashboard=dashboardModel}}`);
+    assert.notOk($('.action.schedule').is(':visible'),
+      'Dashboard schedule button should not be visible');
+
+    config.navi.FEATURES.enableScheduleDashboards = originalFeatureFlag;
+  });
```
Comment (jkusa) [inline packages/dashboards/tests/integration/components/navi-dashboard-test.js]: Can we test a few other things:

- after clicking the `schedule` button, the model opens
- verify you can't schedule a dashboard that isn't owned by the user

Comment (Xuefeng-Zhu) [reply packages/dashboards/tests/integration/components/navi-dashboard-test.js]: > after clicking the schedule button, the model opens
There is already test for this

> verify you can't schedule a dashboard that isn't owned by the user
 The button will not been shown to other users

File: packages/dashboards/tests/integration/components/navi-dashboard-test.js [hunk 2]
```diff
@@ -115,4 +116,36 @@ test('dashboard export', function(assert) {
...
   });
-})
\ No newline at end of file
+});
+
+test('dashboard schedule - config', function (assert) {
+  assert.expect(2);
+  let originalFeatureFlag = config.navi.FEATURES.enableScheduleDashboards;
+
+  return wait().then(() => {
```
Comment (balajimanoharan) [inline packages/dashboards/tests/integration/components/navi-dashboard-test.js]: Looks like `beforeEach` is returning a promise. Is this `wait()` still needed?

File: packages/dashboards/tests/integration/components/navi-dashboard-test.js [hunk 3]
```diff
@@ -115,4 +116,36 @@ test('dashboard export', function(assert) {
...
+      'Dashboard schedule button should not be visible');
+
+    config.navi.FEATURES.enableScheduleDashboards = originalFeatureFlag;
+  });
+});
+
+test('dashboard schedule - isUserOwner', function (assert) {
+  assert.expect(1);
+
+  return wait().then(() => {
```
Comment (balajimanoharan) [inline packages/dashboards/tests/integration/components/navi-dashboard-test.js]: Same here.

File: packages/dashboards/tests/integration/components/navi-dashboard-test.js (hunk 2 above)
Comment (Xuefeng-Zhu) [reply packages/dashboards/tests/integration/components/navi-dashboard-test.js]: It is needed.

File: packages/dashboards/tests/acceptance/schedule-modal-test.js [hunk 4]
```diff
@@ -79,3 +77,18 @@ test('schedule modal save new schedule', function (assert) {
...
+  assert.expect(1);
+  visit('/dashboards/2/view');
+
+  // TriggerEvent does not work here, need to use jquery trigger mouseenter
+  andThen(() => $('.navi-collection__row:first-of-type').trigger('mouseenter'));
+  // Click "Schedule"
+  click('.schedule-action__button');
+
+  andThen(() => {
+    assert.ok(find('.schedule-modal__header .primary-header').is(':visible'),
```
Comment (jkusa) [inline packages/dashboards/tests/acceptance/schedule-modal-test.js]: Can we just check `.schedule-modal`?

File: packages/dashboards/tests/integration/components/navi-dashboard-test.js (hunk 2 above)
Comment (balajimanoharan) [reply packages/dashboards/tests/integration/components/navi-dashboard-test.js]: Oh ok. i missed the setting of the feature flag. Thats probably why this wait is needed.

File: packages/dashboards/tests/integration/components/navi-dashboard-test.js (hunk 3 above)
Comment (balajimanoharan) [reply packages/dashboards/tests/integration/components/navi-dashboard-test.js]: We could probably remove the wait here.

**Pull request merged**